src/pal/src/thread/thread.cpp

```diff
@@ -573,7 +568,22 @@ CorUnix::InternalCreateThread(
         palError = ERROR_INVALID_PARAMETER;
         goto EXIT;
     }
-    
+
+    if (dwStackSize != 0)
+    {
+        // Some systems require the stack size to be aligned to the page size
+        DWORD alignedStackSize = static_cast<DWORD>(ALIGN_UP(dwStackSize, OS_PAGE_SIZE));
+        if (alignedStackSize < dwStackSize)
+        {
+            // When coming here from the public API surface, the incoming value is originally a nonnegative signed int32, so
+            // this shouldn't happen
+            ASSERT("Couldn't align the requested stack size (%u) to the page size\n", dwStackSize);
```

Comment:
Oops, right. So please keep it as it is. But in that case, it seems it would make sense to change the message to say that the problem was the stack size being too large, since it is the only case when the attempt to align would fail.